core/controlplane/config/templates/cloud-config-controller

```diff
@@ -1192,12 +1201,13 @@ write_files:
                       # Fetch the list of instances being terminated by their respective ASGs
                       instances_to_drain=$(asg describe-auto-scaling-groups | jq -r '.AutoScalingGroups[] | select((.Tags[].Key | contains("kube-aws")) and (.Tags[].Key | contains("kubernetes.io/cluster/{{.ClusterName}}"))) | .Instances[] | select(.LifecycleState == "Terminating:Wait") | .InstanceId')
```

Comment:
I don't understand why `node=` bit and patching is needed.  Deploy should just dump all pending ASG termination into configmap, this way it has one owner and daemonsets dont need to clean it up. It also makes it just a single write call to apiserver. 

To me it looks like it can be just:
`
instances_to_drain=$(... | jq  '[.AutoScalingGroups[] | ... | .InstanceId] | sort | join("\n") ' # build json escaped string

cat >yaml  <<EOF
apiVersion: v1
kind: ConfigMap
metadata:
name: kube-node-drainer-status
namespace: kube-system
data:
  asg: ${instances_to_drain}
EOF

# TODO don't apply if $instances_to_drain didn't change since last iteration
kubectl apply -f yaml
```

Then on daemonset side:

```
INSTANCE_ID=$(....)
if [ -e "/etc/kube-node-drainer/from-asg" ] && grep -q $INSTANCE_ID /etc/kube-node-drainer/from-asg; then
   termination_source=asg; break
fi
```
and
```
             - name: kube-node-drainer-status
                projected:
                  sources:
                  - configMap:
                      name: kube-node-drainer-status
                      items:
                       - key: asg
                         path: from-asg
```